.travis/install.sh

```diff
@@ -3,54 +3,80 @@
-    #brew outdated openssl || brew upgrade openssl
-    export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig:$PKG_CONFIG_PATH"
-    brew install readline
-    export PKG_CONFIG_PATH="/usr/local/opt/readline/lib/pkgconfig:$PKG_CONFIG_PATH"
-    brew install zstd
+# Abort installation and don't run tests for pull requests if commit only changed the docs
+if [ "${TRAVIS_EVENT_TYPE}" = "pull_request" ]
+then
+    printf '%s\n' "Checking whether ${TRAVIS_COMMIT_RANGE} changed only docs"
+    git diff --name-only "${TRAVIS_COMMIT_RANGE}" | grep --quiet --invert-match --extended-regexp '(AUTHORS|README\.rst|^(docs)/)' || {
+        printf '%s\n' "Only docs were updated, stopping build process."
+        exit 0
+    }
+fi
+
+# Travis clones with depth=50 which might be too shallow for git describe, see https://github.com/pypa/setuptools_scm/issues/93
+git fetch --unshallow --tags
+
+if [ "${TRAVIS_OS_NAME}" = "osx" ]
+then
+
+    # Update brew itself
+    export HOMEBREW_NO_AUTO_UPDATE=1  # Auto-updating everything would take too much time
+    brew update
+    brew cleanup  # Preempt possible scheduled clean-up so it doesn't clutter the log later
+
+    # Install and/or upgrade dependencies
     brew install lz4
-    brew install xz  # required for python lzma module
-    brew install Caskroom/cask/osxfuse
+    brew install xz  # Required for Python lzma module
+    brew install Caskroom/cask/osxfuse  # Required for Python llfuse module
+    brew install pyenv || brew upgrade pyenv  # Make sure to use most recent pyenv
 
-    brew outdated pyenv || brew upgrade pyenv
-    if which pyenv > /dev/null; then
-        eval "$(pyenv init -)"
-    fi
+    # Configure pkg-config to use OpenSSL 1.1 from Homebrew
+    export PKG_CONFIG_PATH="/usr/local/opt/openssl@1.1/lib/pkgconfig:${PKG_CONFIG_PATH}"
 
-    case "${TOXENV}" in
-        py35)
-            pyenv install 3.5.3  # minimum for openssl 1.1.x
-            pyenv global 3.5.3
-            ;;
-        py37)
-            pyenv install 3.7.0
-            pyenv global 3.7.0
-            ;;
-    esac
+    # Configure pyenv with Python version according to TOXENV
+    eval "$(pyenv init -)"
+    if [ "${TOXENV}" = "py35" ]
+    then
+        pyenv install 3.5.3  # Minimum version for OpenSSL 1.1.x
+        pyenv global 3.5.3
+    elif [ "${TOXENV}" = "py37" ]
+    then
+        pyenv install 3.7.0
+        pyenv global 3.7.0
+    else
+        printf '%s\n' "Unexpected value for TOXENV environment variable"
+        exit 1
+    fi
     pyenv rehash
-    python -m pip install --user virtualenv
-else
-    pip install virtualenv
+
+elif [ "${TRAVIS_OS_NAME}" = "linux" ]
+then
+
+    # Install dependencies
     sudo apt-get update
     sudo apt-get install -y pkg-config fakeroot
-    #sudo apt-get install -y liblz4-dev  # too old on trusty and xenial
-    #sudo apt-get install -y libzstd-dev  # too old on trusty and xenial
```

Comment:
maybe keep at least the information in a comment.

Reply:
Can you explain why the comments are there in the first place?

Reply:
we can use them if they start not being too old in the dist, maybe with bionic.

Reply:
But what advantage do those packages even provide? Are currently tests being skipped? :thinking:

Reply:
don't remember, could be. installing them on linux is almost zero effort, unlike to brew installation.

Reply:
:+1: